src/agent/server.clj

```diff
@@ -0,0 +1,38 @@
+(ns agent.server
+  (:require [agent.secrets :as secrets]
+            [compojure.core :as compojure :refer [GET POST]]
+            [ring.middleware.params :as params]
+            [compojure.route :as route]
+            [aleph.http :as http]))
+
+
+(defn get-secrets-handler [req]
+  (let [result (secrets/get-filesystem-secret {})]
+    (if (first result)
+      {:status 200
+       :headers {"content-type" "application/json"}
+       :body (-> (first result)
+                 (:secrets)
+                 (clojure.data.json/write-str))}
+      {:status 404
+       :headers {"content-type" "application/json"}
+       :body "{\"message\": \"secrets not found\"}"})))
+
+(defn post-secrets-handler [req]
+  (let [body (slurp (:body req))
```

Comment:
I think we could verify if the body is a json type, an error could return if it's not the case (don't know if compojure deals with that)

Reply:
The idea here is to persist whatever comes from the UI, in the format they come (we decided this at some point, but I do not know if we documented it....). This means that we are persisting whatever comes from our UI.

However, an error will occur when trying to "use" this secret, if it is not JSON.

Maybe we should validate it, despite of being a controlled environment (our front end only will call this endpoint).